### ISSUE TITLE
Excluded directories are still scanned

### ISSUE THREAD

**Issue by beatcracker** [megalinter/megalinter#932]
**Describe the bug**
Despite having directories excluded in both `ADDITIONAL_EXCLUDED_DIRECTORIES` and `.gitignore` they're still scanned by some linters.

I've multiple `.wip` (sub)directories in local repo which are excluded by global `.gitignore` rule. These directories should be ignored by linters.

**To Reproduce**

- Windows 10/ Docker + WSL2 backend
- The `npx` is run in Windows, the repo is on local NTFS volume and not in WSL/WSL2.

```shell
npx --yes "mega-linter-runner@^5.0.0" --flavor dotnet
```

- `.mega-linter.yml`
```yaml
ADDITIONAL_EXCLUDED_DIRECTORIES:
  - .wip
APPLY_FIXES: none # all, none, or list of linter keys
DEFAULT_BRANCH: master # Usually master or main
SHOW_ELAPSED_TIME: true
FILEIO_REPORTER: false
LOG_LEVEL: DEBUG
```

**Expected behavior**

Ignored directories should be ignored :).

**Additional context**

Debug log. As you can see `*.cs` is in` .gitignore` and still gets scanned, the `*.key` and `*.yml` files are not detected as ignored but they should be.

- `mega-linter.log`
```none
----------------------------------------------------------------------------------------------------
 - Image Creation Date: 2021-11-04T06:39:13Z
 - Image Revision: a8156288
 - Image Version: v5

...

EXCLUDED_DIRECTORIES=['.wip']

...

- Excluding .gitignored files [2418]: ... , /tmp/lint/tools/.wip/out/wpf/obj/Debug/net5.0-windows/App.g.cs , ... ,  

...

[jscpd] result: 1 Clone found (csharp):
 - tools/.wip/out/wpf/obj/Debug/net5.0-windows/App.g.cs [2:1 - 39:8] (37 lines, 268 tokens)
   tools/.wip/out/wpf/obj/Debug/net5.0-windows/MainWindow.g.cs [2:1 - 39:15]

...

Clone found (yaml):
 - .wip/build/preview.yml [329:5 - 347:27] (18 lines, 118 tokens)
   .wip/build/preview.yml [108:5 - 126:34]

...

[1m### Processed [CREDENTIALS] files[0m
- Using [secretlint v4.1.0] https://github.com/secretlint/secretlint
- Mega-Linter key: [CREDENTIALS_SECRETLINT]
- Rules config: [.secretlintrc.json]
--Error detail:

[4m/tmp/lint/.wip/foobar.key[24m
  [90m5:18[39m  [31merror[39m  [PrivateKey] found private key: -----BEGIN PRIVATE 
```

**Notes**

I also observe `misspell` scanning `.git` directory, which doesn't make much sense:

```none
[1m### Processed [SPELL] files[0m
- Using [misspell v0.3.4] https://github.com/client9/misspell
- Mega-Linter key: [SPELL_MISSPELL]
- Rules config: identified by [misspell]
[misspell] .cspell.json
[misspell] .git/COMMIT_EDITMSG
[misspell] .git/FETCH_HEAD

...

[misspell] .git/logs/refs/remotes/origin/test_replace_stages_with_extends
[misspell] .git/logs/refs/stash
[misspell] .git/objects/00/00ab99ffdd8fd52ab8759029bb06bdf2b15f25

...
```

**Comment by nvuillam** [comment]
Linters that scan in "project" mode (on root directory) are ignoring MegaLinter filtering configurations

Are you using v5 ? If yes, IGNORE_GITIGNORED_FILES is set to true by default

**Comment by beatcracker** [comment]
It's v5, yes. I confess that I'm a little confused by the configuration docs, since there is

- `[ADDITIONAL_]EXCLUDED_DIRECTORIES`
- `FILTER_REGEX_[INCLUDE|EXCLUDE]`

and I'm not exactly sure how they relate to each other. Plus it's not very clear, which linters run in `"project" mode`.

It would be nice if we had this information in [supported-linters](https://megalinter.github.io/supported-linters/) table, otherwise one needs to check every linter docs to figure out how to apply configuration consistently.

**Comment by nvuillam** [comment]
Would you like to share your repo so we can have a look ? :)~~~

**Comment by llaville** [comment]
@nvuillam I confirm, in another situation, that `eslint-plugin-jsonc` linter do not follow ignore rules condition !

With project https://github.com/llaville/php-compatinfo-db, not yet fully migrated to megalinter v5 (remotely)

When I run it locally with a fresh documentation flavor docker image
```
----------------------------------------------------------------------------------------------------
------------------------------------------- Mega-Linter --------------------------------------------
----------------------------------------------------------------------------------------------------
 - Image Creation Date: 2021-11-04T06:38:55Z
 - Image Revision: a8156288
 - Image Version: v5
```
with following command: `time docker run --rm -v $(pwd):/tmp/lint megalinter/megalinter-documentation:v5`, I got unexpected results.

![eslint_plugin_issue](https://user-images.githubusercontent.com/364342/140651090-cd632b05-d389-46a0-821e-e7ab3c03b490.png)

```
### Processed [JSON] files
- Using [eslint-plugin-jsonc v1.7.0] https://ota-meshi.github.io/eslint-plugin-jsonc/
- Mega-Linter key: [JSON_ESLINT_PLUGIN_JSONC]
- Rules config: [.eslintrc-json.json]
[eslint-plugin-jsonc] .github/linters/.markdown-lint.json
[eslint-plugin-jsonc] composer.json
[eslint-plugin-jsonc] data/reference/extension/amqp/0/classes.json

... and much MORE ...

[eslint-plugin-jsonc] vendor/symfony/var-exporter/composer.json
--Error detail:

/tmp/lint/vendor/composer/composer/doc/fixtures/repo-composer-with-providers/p/bar/baz$923363b3c22e73abb2e3fd891c8156dd4d0821a97fd3e428bc910833e3e46dbe.json

```
While `vendor/` folder is included in `.gitignore`  see https://github.com/llaville/php-compatinfo-db/blob/master/.gitignore#L6

With https://github.com/llaville/php-compatinfo-db/blob/master/.mega-linter.yml#L4

**Comment by llaville** [comment]
Regression found since PR #837; See https://github.com/megalinter/megalinter/blob/v5.0.7/megalinter/MegaLinter.py#L313, returns a string, not a boolean
See good syntax at https://github.com/megalinter/megalinter/blob/v5.0.7/megalinter/MegaLinter.py#L317

If you want to have default true value, then put 

```
if config.exists("IGNORE_GITIGNORED_FILES"):
            self.ignore_gitignore_files = config.get("IGNORE_GITIGNORED_FILES", "true") == "true"
```

**Comment by nvuillam** [comment]
@llaville please can you upgrade https://github.com/llaville/php-compatinfo-db, to v5 and see if there is still the problem ?

Run `mega-linter-runner --upgrade` and it should be ok :)

**Comment by beatcracker** [comment]
> @nvuillam 
> Would you like to share your repo so we can have a look ? :)~~~

Alas, no, `$DAYJOB` stuff. But I probably can make something akin to it to repro.

**Comment by nvuillam** [comment]
@beatcracker if you can use **beta** version of v5 (megalinter/megalinter@beta ) you'll have the version containing @llaville 's updates

**Comment by llaville** [comment]
@nvuillam 👍 for beta that should again ignore files/directories as expected 

```
----------------------------------------------------------------------------------------------------
------------------------------------------- Mega-Linter --------------------------------------------
----------------------------------------------------------------------------------------------------
 - Image Creation Date: 2021-11-08T00:13:36Z
 - Image Revision: 63097f02
 - Image Version: beta
```
